### PR TITLE
fix: extend ARM platform changes through to deployment

### PR DESCRIPTION
**Pull request by huanjani**
These changes enable workload images on ARM machines to build as amd64 images.
If the underlying docker engine detected is ARM architecture-based, Copilot builds with `--platform linux/amd64` so workloads deploy rather than going through the whole workflow only to fail to spin up. 

Related: #2636.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License..

### CODE REVIEW

File: internal/pkg/cli/job_init_test.go
Comment (efekarakus) [inline internal/pkg/cli/job_init_test.go]: nit:
```suggestion
						Platform:       &manifest.PlatformArgsOrString{},
```

File: internal/pkg/cli/svc_init_test.go
Comment (efekarakus) [inline internal/pkg/cli/svc_init_test.go]: samesies here

File: internal/pkg/manifest/rd_web_svc.go
Comment (efekarakus) [inline internal/pkg/manifest/rd_web_svc.go]: I'm trying to remember why we even allow instance config to have a Platform for RDWS? since it only supports linux arm64

File: internal/pkg/template/templates/workloads/jobs/scheduled-job/manifest.yml
Comment (efekarakus) [inline internal/pkg/template/templates/workloads/jobs/scheduled-job/manifest.yml:37]: There is a hidden assumption here that the `.Platform` field is always non-nil. This is true because of the `cli` pkg but it's not obvious. These templates are rendered by the `manifest` pkg so it should be its responsibility to ensure that `Platform` always has a value.  


Can we maybe add this behavior in the default constructors?
```go
newDefaultBackendService() {
   ...
   BackendServiceConfig: BackendServiceConfig {
       ...,
       Platform: &PlatformArgsOrString{}, // Ensure .Platform is always set.
   }
}
```

File: internal/pkg/manifest/rd_web_svc.go
Comment (huanjani) [reply internal/pkg/manifest/rd_web_svc.go]: Took it out of `RequestDrivenWebServiceProps` bc it won't go in manifests, but don't we need it in InstanceConfig for BuildArgs?

File: internal/pkg/manifest/backend_svc.go
Comment (iamhopaul123) [inline internal/pkg/manifest/backend_svc.go]: Is it equivalent to
```suggestion
	svc.BackendServiceConfig.Platform = props.Platform
```
right now?

File: internal/pkg/template/templates/workloads/jobs/scheduled-job/manifest.yml
Comment (iamhopaul123) [reply internal/pkg/template/templates/workloads/jobs/scheduled-job/manifest.yml:37]: Why don't we set it to be struct instead of pointer of struct?

File: internal/pkg/manifest/rd_web_svc.go
Comment (huanjani) [reply internal/pkg/manifest/rd_web_svc.go]: Won't we need this field if `ARM` is redirecting to `amd64` for `BuildArgs`?

File: internal/pkg/template/templates/workloads/jobs/scheduled-job/manifest.yml
Comment (huanjani) [reply internal/pkg/template/templates/workloads/jobs/scheduled-job/manifest.yml:37]: It's small, but for space considerations? We have other pointers to structs for other default values, too.

File: internal/pkg/manifest/backend_svc.go
Comment (huanjani) [reply internal/pkg/manifest/backend_svc.go]: Without the if-statement, the default behavior (when there isn't a platform redirection) would make the whole Platform field nil, not just the member fields. So if we want the field to always be set as default, then we should leave the if-statement in.

File: internal/pkg/template/templates/workloads/jobs/scheduled-job/manifest.yml
Comment (iamhopaul123) [reply internal/pkg/template/templates/workloads/jobs/scheduled-job/manifest.yml:37]: It's not small now lol. we are now trying to make it consistent to always use non pointer unless it is necessary. For this one using non pointer we can be sure that .Platform is always set.

File: internal/pkg/manifest/backend_svc.go
Comment (iamhopaul123) [reply internal/pkg/manifest/backend_svc.go]: but what's the difference between platform and [healthcheck](https://github.com/aws/copilot-cli/pull/2814/files#diff-2147c7bfea12eb3dff6a866c245ac1ed73e671d7c47bf06ff906e9dfc741d380R55)?

> the default behavior (when there isn't a platform redirection) would make the whole Platform field nil, not just the member fields.

But this is a constructor, `svc.BackendServiceConfig.Platform` should be initially nil right (or empty struct if you remove the pointer)?

File: internal/pkg/cli/job_init.go
Comment (efekarakus) [inline internal/pkg/cli/job_init.go]: nit: maybe
```suggestion
		log.Warningf("Your architecture type is currently unsupported. Setting 'platform' to %s instead in your manifest.\n", dockerengine.DockerBuildPlatform(dockerengine.LinuxOS, dockerengine.Amd64Arch))
```

File: internal/pkg/manifest/applyenv_test.go
Comment (efekarakus) [inline internal/pkg/manifest/applyenv_test.go]: I think this test was deleted now in mainline 😅

File: internal/pkg/manifest/backend_svc.go
Comment (efekarakus) [inline internal/pkg/manifest/backend_svc.go]: no need to define it here now since it's no longer a pointer, the zero value `PlatformArgsOrString{}`

File: internal/pkg/manifest/job.go
Comment (efekarakus) [inline internal/pkg/manifest/job.go]: same here, we can remove this field from all constructors :)

**Review by efekarakus — COMMENTED** [review]
Looks great! just small refactor to remove the field from the constructor.

I'll approve :shipit: after mainline is merged!

File: internal/pkg/manifest/backend_svc_test.go
Comment (iamhopaul123) [inline internal/pkg/manifest/backend_svc_test.go]: same here...

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/cli/job_init.go
Comment (huanjani) [reply internal/pkg/cli/job_init.go]: Okay, but I'll leave the wording as is in `svc_init` because of the platform field not being surfaced in RDWS.

**Review by efekarakus — APPROVED** [review]
pew pew :shipit:

**Pull request merged**
